### PR TITLE
[dhctl] disable deckhouse init container if external-module-manager if off

### PR DESCRIPTION
**Pull request by mvasl**
## Description

Added pod security context to deckhouse pod and removed external modules init container from it if `external-module-manager` is disabled.

## Why do we need it, and what problem does it solve?

This fixes inconsistent helm manifests during initial bootstrap

## Checklist
- [ ] The code is covered by unit tests.
- [x] e2e tests passed.
- [ ] Documentation updated according to the changes.
- [ ] Changes were tested in the Kubernetes cluster manually.

## Changelog entries

```changes
section: dhctl
type: fix
summary: Added pod security context to deckhouse pod and disabled deckhouse init container if `external-module-manager` if off
impact_level: default
```

### CODE REVIEW

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @mvasl. I've started the workflow for label 'e2e/run/azure'...

**Comment by mvasl** [comment]
/e2e/abort azure;Standard;containerd;1.25 8723171594 8723171594-1-con-1-25 /sys/deckhouse-oss/install:pr8173 azureuser@40.113.160.141

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @mvasl. I've recognized your '/e2e/abort' command and started the workflow...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @mvasl. I've started the workflow for label 'e2e/run/openstack'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @mvasl. I've recognized your '/e2e/abort' command and started the workflow...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @mvasl. I've started the workflow for label 'e2e/run/openstack'...

**Comment by mvasl** [comment]
/e2e/abort openstack;Standard;containerd;1.25 8725245759 8725245759-1-con-1-25 /sys/deckhouse-oss/install:pr8173 redos@46.161.53.212

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @mvasl. I've recognized your '/e2e/abort' command and started the workflow...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @mvasl. I've started the workflow for label 'e2e/run/azure'...

**Comment by mvasl** [comment]
/e2e/abort azure;Standard;containerd;1.25 8737100697 8737100697-1-con-1-25 /sys/deckhouse-oss/install:pr8173 azureuser@4.245.4.180

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @mvasl. I've recognized your '/e2e/abort' command and started the workflow...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @mvasl. I've started the workflow for label 'e2e/run/gcp'...

File: dhctl/pkg/kubernetes/actions/deckhouse/install.go [hunk 1]
```diff
@@ -592,13 +592,23 @@ func CreateDeckhouseDeployment(kubeCl *client.KubernetesClient, cfg *config.Deck
 }
 
 func deckhouseDeploymentParamsFromCfg(cfg *config.DeckhouseInstaller) manifests.DeckhouseDeploymentParams {
+	externalModuleManagerEnabled := true
+	for _, mc := range cfg.ModuleConfigs {
```
[the remark this answers is not in the document]
Comment (mvasl) [reply dhctl/pkg/kubernetes/actions/deckhouse/install.go:596]: I was told it will

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @mvasl. I've started the workflow for label 'e2e/run/gcp'...

File: dhctl/pkg/kubernetes/actions/deckhouse/install.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (mvasl) [reply dhctl/pkg/kubernetes/actions/deckhouse/install.go:597]: Figured we should temporarily add a check for minimal bundles here, until external modules manager is integrated into deckhouse-controller, then remove it all at once

**Review by name212 — APPROVED** [review]

**Review by name212 — APPROVED** [review]

**Comment by deckhouse-BOaTswain** [comment]
Failure: cherry-pick commit 440f18bb196d8f398931e8988a25a7393de30e77 to the branch [release-1.59](https://github.com/deckhouse/deckhouse/tree/release-1.59) failed. See [Job](https://github.com/deckhouse/deckhouse/actions/runs/8752241098) for details.

**Comment by z9r5** [comment]
Cherry-picked into release-1.59

**Pull request merged**